pkg/config/piped.go

```diff
@@ -919,13 +926,35 @@ func (n *NotificationReceiver) Mask() {
 }
 
 type NotificationReceiverSlack struct {
-	HookURL string `json:"hookURL"`
+	HookURL    string `json:"hookURL"`
+	OAuthToken string `json:"oauthToken"`
+	ChannelID  string `json:"channelID"`
 }
 
 func (n *NotificationReceiverSlack) Mask() {
 	if len(n.HookURL) != 0 {
 		n.HookURL = maskString
 	}
+	if len(n.OAuthToken) != 0 {
+		n.OAuthToken = maskString
+	}
+}
+
+func (n *NotificationReceiverSlack) Validate() error {
+	if n.HookURL != "" {
+		if n.OAuthToken != "" || n.ChannelID != "" {
+			return errors.New("slack notification receiver: hookURL is set, so oauthToken and channelID should be empty")
+		}
+	}
+	if n.OAuthToken != "" {
+		if n.HookURL != "" {
+			return errors.New("slack notification receiver: oauthToken is set, so hookURL should be empty")
+		}
+		if n.ChannelID == "" {
+			return errors.New("slack notification receiver: oauthToken is set, so channelID should not be empty")
+		}
+	}
+	return nil
```

Comment:
```suggestion
	if n.HookURL != "" && n.OAuthToken != "" {
		return errors.New("only one of hookURL or oauthToken and channelID should be used")
	}
	
	if n.HookURL != "" {
		return nil
	}
	
	if n.OAuthToken == "" || n.ChannelID == "" {
		return errors.New("missing oauthToken or channelID configuration")
	}
	return nil
```
nits, how about this?

Reply:
```go
	if n.HookURL != "" && n.OAuthToken != "" {
		return errors.New("only one of hookURL or oauthToken and channelID should be used")
	}
```
If HookURL and ChannelID is set, is this configuration valid ?

```go
	if n.OAuthToken == "" || n.ChannelID == "" {
		return errors.New("missing oauthToken or channelID configuration")
	}
```
👍

Reply:
@khanhtc1202 
wdyt?

Reply:
I see, then, it should be

```golang
	if n.HookURL != "" && (n.OAuthToken != "" || n.ChannelID == "") {
```